e2etest/l2tests/l2.go

```diff
@@ -232,6 +232,54 @@ var _ = ginkgo.Describe("L2", func() {
 			err = wget.Do(address, executor.Host)
 			framework.ExpectNoError(err)
 		})
+
+		ginkgo.It("should not be announced from a node with a NetworkUnavailable condition", func() {
```

Comment:
since the choosen node is random, I'd leverage the node selector feature to pin the announcement to one node, then check the service is not advertised, and then restore the condition back.
Here we won't know if the test passes because we are lucky or because it works.

Reply:
If I understand correctly, you are concerned that in this L2 test, the announcer node could change due to a random election. i.e. the first node which we mark as "nodeToSet" wouldn't get elected again after changing back the condition to false.
I don't think the election is random in the sense of we could have a different leader in both elections where all the nodes are available.
The elections are based on the hash of the nodes name and IP and those do not change during this test.
correct me if I'm wrong.

Reply:
yep, I mis-read the behaviour.